libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -186,12 +186,18 @@ func (n *bridgeNetwork) addPortMappings(
 			if !ok {
 				return nil, fmt.Errorf("invalid child host IP address %s in %s", b.childHostIP, b)
 			}
-			b.portDriverRemove, err = pdc.AddPort(ctx, b.Proto.String(), hip, chip, int(b.HostPort))
+			bindings[i].portDriverRemove, err = pdc.AddPort(ctx, b.Proto.String(), hip, chip, int(b.HostPort))
 			if err != nil {
 				return nil, err
 			}
 		}
-		if err := n.setPerPortIptables(bindings[i], true); err != nil {
+		if err := n.setPerPortIptables(b, true); err != nil {
+			return nil, err
+		}
+		// Even when NAT isn't used (eg. when an IPv6 HostIP is specified, but
+		// the container has no IPv6 address), the container's port must be
+		// protected by filterPortByInputIface.
```

Comment:
At first glance, this comment appears to conflict with the one on line 901 ("NAT is disabled -- no need to filter by input interface" -> early return).

I think this one's about `docker-proxy` - nat's happening, in the sense that there's a host port (on an IPv6 address) getting proxied to some port on a container's IPv4 address, it's just not iptables-NAT.

The other comment is about "routed mode", where there's just no host port to map from.